QuantConnect.ExanteBrokerage/ExanteSymbolMapper.cs

```diff
@@ -0,0 +1,303 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Brokerages;
+using Log = QuantConnect.Logging.Log;
+using System.Linq;
+using QuantConnect.Util;
+
+namespace QuantConnect.ExanteBrokerage
+{
+    public static class ExanteMarket
+    {
+        public const string NASDAQ = "NASDAQ";
+        public const string ARCA = "ARCA";
+        public const string AMEX = "AMEX";
+        public const string EXANTE = "EXANTE";
+        public const string USD = "USD";
+        public const string USCORP = "USCORP";
+        public const string EUR = "EUR";
+        public const string GBP = "GBP";
+        public const string ASN = "ASN";
+        public const string CAD = "CAD";
+        public const string AUD = "AUD";
+        public const string ARG = "ARG";
+        public const string OTCMKTS = "OTCMKTS";
+    }
+
+    public class ExanteSymbolLocal
+    {
+        public string Exchange { get; }
+        public string SymbolId { get; }
+
+        public ExanteSymbolLocal(string exchange, string symbolId)
+        {
+            Exchange = exchange;
+            SymbolId = symbolId;
+        }
+
+        public string Id => $"{SymbolId}.{Exchange}";
+    }
+
+    /// <summary>
+    /// Provides the mapping between Lean symbols and Exante symbols.
+    /// </summary>
+    public class ExanteSymbolMapper : ISymbolMapper
+    {
+        private readonly Dictionary<string, string> _leanSymbolIdToExanteExchange;
+        private readonly ExanteClientWrapper _client;
+        private readonly HashSet<string> _supportedCryptoCurrencies;
+
+        public ExanteSymbolMapper(
+            ExanteClientWrapper client,
+            HashSet<string> supportedCryptoCurrencies)
```

Comment:
I think we should not hard code any ticker, they will get outdated really fast. Could we replace this by an API call? Or just let it fail when trying to add a currency that isn't supported at runtime?
Symbols API
![image](https://user-images.githubusercontent.com/18473240/158389103-19616a87-99c4-4685-9395-e8dc501c4315.png)

Reply:
I tried the approach you suggest. There are problems as follows. There are cryptos:

![image](https://user-images.githubusercontent.com/574267/159991674-39563e33-4c6b-4dba-a4fd-c74f66f4bc68.png)

But those don't have explicit exchange in ticker:

![image](https://user-images.githubusercontent.com/574267/159991793-55e604ce-f526-48da-8a6d-5148713861a9.png)

Though, I can't filter symbols by exchange. Some time ago cryptos were under `ExanteSymbolType.Fund` as stated at https://github.com/d-ugarov/Exante.Net/issues/1. But Exante moved it from that type. And I don't know where are they now. 

In total, I don't know using which properties to filter all the symbols.

Reply:
I asked Exante support once again. They just don't answer how to get the list via API, but simply send me to REST API documentation.

Experimentally I found out the code to fetch the exact same list:

```c#
List<ExanteSymbol> cryptoCurrencies =
                    _exanteClient
                        .GetSymbolsByTypeAsync(ExanteSymbolType.CFD)
                        .SynchronouslyAwaitTaskResult()
                        .Data
                        .Where(x => x.Exchange == "USD")
                        .ToList();
```

Previously they were under `Fund` symbol type. I don't know why it under `CFD` and with `USD` `Exchange`. Either don't know how long it would be there. Do you think I should replace hard-coded list with that code?

Reply:
CFD sounds like [Contract for Differences](https://www.investopedia.com/terms/c/contractfordifferences.asp) which for Lean is it's own asset type with it's own contract specifications (can't trade CFDs in the US. Can see examples at https://github.com/QuantConnect/Lean/blob/master/Data/symbol-properties/symbol-properties-database.csv . Is that the case for Exante? Seems like the market would be Exante for those assets like in those examples and we don't really need any list in the brokerage implementation?

Reply:
@Martin-Molinero Martin, the Exante support would not understand the question about eligibility of trading CFDs in USA. If you somehow related to US, please check if there are crypto currencies available in terminal in demo mode.

Reply:
Suggest not supporting crypto trading on exante for now and ship the rest first, which is a lot already, we can go back and review it in depth 👍

Reply:
Fixed by 3580b3af440564c934575b3b988bebcbd122886e